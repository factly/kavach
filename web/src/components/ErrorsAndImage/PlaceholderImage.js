import React from 'react';
import placeholderImg from './placeholderImage.svg';
function PlaceholderImage({ width, height, maxWidth }) {
  return (
    <>
      <img
        src={placeholderImg}
        width={width}
        height={height}
        alt=""
        style={{ objectFit: 'contain', padding: '1rem', maxWidth: maxWidth || '240px' }}
      />
    </>
  );
}

PlaceholderImage.defaultProps = {
  height: 'auto',
  width: '100%',
};

export default PlaceholderImage;
