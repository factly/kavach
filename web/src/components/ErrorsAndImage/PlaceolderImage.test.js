import React from "react";
import PlaceholderImage from "./PlaceholderImage";
import { render, shallow } from "enzyme";

describe("PlaceholderImage", () => {
	it("should render the component", () => {
		const props = { width: "100%", height: "auto", maxWidth: "240px" };
		let component = shallow(<PlaceholderImage {...props} />);
		expect(component).toMatchSnapshot();
	});

	it("should render the component when maxWidth undefined", () => {
		const props = { width: "100%", height: "auto" };
		let component = render(<PlaceholderImage {...props} />);
		expect(component).toMatchSnapshot();
		expect(component.attr("style")).toEqual("object-fit:contain;padding:1rem;max-width:240px");
	});
});
