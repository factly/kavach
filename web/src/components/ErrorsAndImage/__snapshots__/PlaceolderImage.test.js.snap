// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderImage should render the component 1`] = `
<Fragment>
  <img
    alt=""
    height="auto"
    src="placeholderImage.svg"
    style={
      Object {
        "maxWidth": "240px",
        "objectFit": "contain",
        "padding": "1rem",
      }
    }
    width="100%"
  />
</Fragment>
`;

exports[`PlaceholderImage should render the component when maxWidth undefined 1`] = `
<img
  alt=""
  height="auto"
  src="placeholderImage.svg"
  style="object-fit:contain;padding:1rem;max-width:240px"
  width="100%"
/>
`;
